# Data Source
https://www.kaggle.com/ankitverma2010/ecommerce-customer-churn-analysis-and-prediction

### who created the data?
ANKIT VERMA

### when created the data?
Date created:
2021-01-26
<br>
Current version:
Version 1

### About this file

Kumpulan data milik perusahaan ECommerce online terkemuka. Sebuah perusahaan ingin mengetahui pelanggan mana yang akan churn, sehingga mereka dapat mendekati pelanggan untuk menawarkan berbagai macam promo.

# Import Data

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("data/ECommerceDataset.csv")
df.head()

CustomerID  Churn  Tenure PreferredLoginDevice  CityTier  WarehouseToHome  \
0       50001      1     4.0         Mobile Phone         3              6.0   
1       50002      1     NaN                Phone         1              8.0   
2       50003      1     NaN                Phone         1             30.0   
3       50004      1     0.0                Phone         3             15.0   
4       50005      1     0.0                Phone         1             12.0   

  PreferredPaymentMode  Gender  HourSpendOnApp  NumberOfDeviceRegistered  \
0           Debit Card  Female             3.0                         3   
1                  UPI    Male             3.0                         4   
2           Debit Card    Male             2.0                         4   
3           Debit Card    Male             2.0                         4   
4                   CC    Male             NaN                         3   

     PreferedOrderCat  SatisfactionScore MaritalStatus  NumberOfAddress  \
0  Laptop & Accessory                  2        Single                9   
1              Mobile                  3        Single                7   
2              Mobile                  3        Single                6   
3  Laptop & Accessory                  5        Single                8   
4              Mobile                  5        Single                3   

   Complain  OrderAmountHikeFromlastYear  CouponUsed  OrderCount  \
0         1                         11.0         1.0         1.0   
1         1                         15.0         0.0         1.0   
2         1                         14.0         0.0         1.0   
3         0                         23.0         0.0         1.0   
4         0                         11.0         1.0         1.0   

   DaySinceLastOrder  CashbackAmount  
0                5.0             160  
1                0.0             121  
2                3.0             120  
3                3.0             134  
4                3.0             130

# Understanding the dataset

Kita memiliki total dua puluh kolom dari dataset.

Deskripsi Kolom :

    1.  CustomerID                  : Nomor identitas pengguna
    2.  Churn                       : Identifikasi pengguna churn atau tidak.
                                        Jika bernilai 1 maka pengguna churn,
                                        Jika bernilai 0 maka pengguna tidak churn    
    3.  Tenure                      : Masa pengguna berlangganan pada produk ini
    4.  PreferredLoginDevice        : Perangkat yang sering digunakan untuk login
    5.  CityTier                    : Klasifikasi kota berdasarkan tingkat kesejahteraan di kota tersebut
                                        1 = daerah Perkotaan
                                        2 = daerah Pedesaan
                                        3 = daerah Tertinggal
    6.  WarehouseToHome             : Jarak antara gudang dan rumah pengguna
    7.  PreferredPaymentMode        : Metode pembayaran yang sering digunakan untuk pengguna
    8.  Gender                      : Jenis kelamin pengguna
    9.  HourSpendOnApp              : Jam yang dihabiskan untuk menggunakan aplikasi atau website
    10. NumberOfDeviceRegistered    : Jumlah perangkat yang diregistrasikan terhadap satu pengguna/akun
    11. PreferredOrderCat           : Kategori yang sering dipesan dalam satu bulan terakhir
    12. SatisfactionScore           : Nilai kepuasan pelanggan terhadap pelayanan
    13. MaritalStatus               : Status pernikahan pelanggan
    14. NumberOfAddress             : Jumlah alamat yang terdaftar dalam satu pengguna
    15. Complain                    : Keluhan yang diajukan dalam satu bulan terakhir
    16. OrderAmountHikeFromLastYear : Persentase peningkatan pesanan dalam satu tahun terakhir
    17. CouponUsed                  : Jumlah kupon yang digunakan dalam satu bulan terakhir
    18. OrderCount                  : Jumlah pesanan dalam satu bulan terakhir
    19. DaySinceLastOrder           : Hari terakhir pemesanan yang dilakukan oleh pelanggan 
    20. CashbackAmount              : Rata-rata cashback dalam satu bulan terakhir 

In [3]:
df.shape

(5630, 20)

Berdasarkan dataset, terdapat 5630 observasi. Kolom churn juga menunjukkan apakah pelanggan akan churn atau tidak.

Kesimpulannya, variabel target adalah churn, dan semua kolom lainnya menjadi variabel fitur.

Setelah deskripsi kolom dan jumlah observasi ditetapkan, mari kita pelajari datasetnya. :)

### Column Types

Dataset ini sedikit rumit karena menampilkan data kategorikal dan kontinu. Jadi, kita perlu memperlakukan kolom secara berbeda berdasarkan jenisnya. Agar model berfungsi, data hanya perlu numerik. Di bawah ini adalah kolom beserta tipe datanya.

In [4]:
df.dtypes

CustomerID                       int64
Churn                            int64
Tenure                         float64
PreferredLoginDevice            object
CityTier                         int64
WarehouseToHome                float64
PreferredPaymentMode            object
Gender                          object
HourSpendOnApp                 float64
NumberOfDeviceRegistered         int64
PreferedOrderCat                object
SatisfactionScore                int64
MaritalStatus                   object
NumberOfAddress                  int64
Complain                         int64
OrderAmountHikeFromlastYear    float64
CouponUsed                     float64
OrderCount                     float64
DaySinceLastOrder              float64
CashbackAmount                   int64
dtype: object

### Feature

Fitur mengacu pada kolom pengamatan tidak termasuk variabel target.<br>
Setiap machine learning model menggunakan fitur untuk prediksi, namun tidak semua fitur dapat digunakan.<br>
Misalnya, ada beberapa kolom yang secara statistik tidak memengaruhi variabel target.<br>
Penghapusan fitur yang tidak perlu membantu meningkatkan kinerja model dan menghemat waktu dan memori saat melatih model.

Dalam hal ini, hal termudah untuk dilakukan adalah menjadikan kolom CustomerID sebagai index kolom karena kolom CustomerID bukan fitur dan ini tidak bermanfaat untuk mesin dalam melatih model.